services/cd-service/pkg/repository/transformer.go

```diff
@@ -101,6 +101,9 @@ func manifestDirectoryWithReleasesVersion(fs billy.Filesystem, application strin
 }
 
 func commitDirectory(fs billy.Filesystem, commit string) string {
+	if !valid.SHA1CommitID(commit) {
+		return fs.Join("commits", "")
```

Comment:
Should we return something different? Verification seemed missing here as invalid use of this function could lead to backend crashing

Reply:
I think at this point we need to rely that the caller already checked this.
Either that or we need to return an error here, but that complicates things.